tests/test_tensorflow_sparse_data_api.py

```diff
@@ -23,122 +21,146 @@
 ROWS = 1000
 
 
-@pytest.fixture(
-    params=[
-        {
-            "input_shape": (10,),
-        },
-    ]
+@pytest.mark.parametrize(
+    "input_shape",
+    [
+        (10,),
+    ],
+)
+# We test for single and multiple attributes
+@pytest.mark.parametrize(
```

Comment:
Also a general comment. Is it necessary for the number of attributes to be given explicitly by the user? Couldn't we somehow infer it by the data during ingestion and accordingly reduce not only the `num_of_attributes` arg but also the `for` loops that are all around following it?

Reply:
I was also thinking of this but there is always the case where a user might need to use only some of the attributes that live in a TileDB array. That's why I thought that passing them explicitly by attribute name should be useful. Any other ideas on this? Maybe if no attributes are passed return them all? Of course it adds some extra complexity.

Reply:
Okay let's just add the case where possibly we want all attributes without the need to explicitly add them one by one.